.mdn-spec-links.ts

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * Each object for a feature has data from the MDN article for the feature,
+ * along with data from BCD https://github.com/mdn/browser-compat-data/.
+ */
+export interface Feature {
+    /**
+     * Name of the feature (interface, method, or property name); from BCD.
+     */
+    name: string;
+    /**
+     * Filename of the BCD file containing the data for the feature.
+     */
+    filename: string | null;
+    /**
+     * Title of the MDN article for the feature.
+     */
+    title: string;
+    /**
+     * Slug (URL path) of the MDN article for the feature.
+     */
+    slug: string;
+    /**
+     * First paragraph or "seoSummary" of the MDN article for the feature.
+     */
+    summary: string;
+    /**
+     * Names of engines with support for the feature that conforms to the
+     * spec (that is, un-prefixed and not under an alternative name —
+     * though possibly requiring a runtime flag or preference to be set).
+     */
+    engines: EngineNames[];
+    /**
+     * Names of engines with support for the feature implemented under a
+     * wholly different name (in contrast to just being prefixed).
+     */
+    altname?: EngineNames[];
+    /**
+     * Names of engines with support for the feature that requires setting
+     * a runtime flag or preference.
+     */
+    needsflag?: EngineNames[];
+    /**
+     * Names of engines with support for the feature that differs from the
+     * spec in a way that may cause compatibility problems.
+     */
+    partial?: EngineNames[];
+    /**
+     * Names of engines with support for the feature that requires adding
+     * a prefix to the standard feature name defined in the spec.
+     */
+    prefixed?: EngineNames[];
+    /**
+     * Per-browser data about support for the feature.
+     */
+    support: SupportBlock | null;
+    /**
+     * The caniuse.com shortname for the feature, along with the title of
+     * the caniuse.com table (page) for the feature.
+     */
+    caniuse?:  { feature: string, title: string };
+}
+
+export type EngineNames = "blink" | "gecko" | "webkit";
+
+export type SupportDetails = SupportStatement | SupportStatement[];
+
+export interface SupportBlock {
+    chrome?:                  SupportDetails;
+    chrome_android?:          SupportDetails;
+    edge?:                    SupportDetails;
+    edge_blink?:              SupportDetails;
+    firefox?:                 SupportDetails;
+    firefox_android?:         SupportDetails;
+    ie?:                      SupportDetails;
+    nodejs?:                  SupportDetails;
+    opera?:                   SupportDetails;
+    opera_android?:           SupportDetails;
+    qq_android?:              SupportDetails;
+    safari?:                  SupportDetails;
+    safari_ios?:              SupportDetails;
+    samsunginternet_android?: SupportDetails;
+    uc_android?:              SupportDetails;
+    uc_chinese_android?:      SupportDetails;
+    webview_android?:         SupportDetails;
+}
```

Comment:
Nit: Maybe we can reduce duplication here:
```ts
type Browsers = "chrome" | "chrome_android" | "edge" // ...
export type SupportBlock = Partial<Record<Browsers, SupportDetails>>;
```

Reply:
> ```ts
> type Browsers = "chrome" | "chrome_android" | "edge" // ...
> export type SupportBlock = Partial<Record<Browsers, SupportDetails>>;
> ```

Great stuff — thanks much for taking time to suggest that refinement. The duplication had seemed really suboptimal to me, but I don’t yet know enough about TypeScript to have realized what could be used to deal with it. (I didn’t yet know about either the Partial or Record types, so I’m glad this gave me an opportunity to learn.)